examples/i-inverse-kinematics.cpp

```diff
@@ -10,37 +10,35 @@ int main(int argc, char ** argv)
   pinocchio::Data data(model);
 
   const int    JOINT_ID = 6;
-  Eigen::Vector3d xdes;        xdes << 0.5, -0.5, 0.5;  
+  Eigen::Vector3d xdes;        xdes << 0.5, -0.5, 0.5;
 
   Eigen::VectorXd q     = pinocchio::neutral(model);
   const double eps      = 1e-4;
-  const int IT_MAX      = 1000
+  const int IT_MAX      = 1000;
   const double DT       = 1e-1;
 
   pinocchio::Data::Matrix6x J(6,model.nv); J.setZero();
   unsigned int svdOptions = Eigen::ComputeThinU | Eigen::ComputeThinV;
 
   for (int i=0 ; i<IT_MAX ; ++i)
+  {
+    pinocchio::forwardKinematics(model,data,q);
+    const Eigen::Vector3d & x   = data.oMi[JOINT_ID].translation();
+    const Eigen::Matrix3d & R   = data.oMi[JOINT_ID].rotation();
+    const Eigen::Vector3d & err = R.transpose()*(x-xdes);
+    if(err.norm() < eps)
     {
-      pinocchio::forwardKinematics(model,data,q);
-      const Eigen::Vector3d & x   = data.oMi[JOINT_ID].translation();
-      const Eigen::Matrix3d & R   = data.oMi[JOINT_ID].rotation();
-      const Eigen::Vector3d & err = R.transpose()*(x-xdes);
-      if(err.norm() < eps)
-        {
-          std::cout << "Convergence achieved!" << std::endl;
-          break;
-        }
-      pinocchio::jointJacobian(model,data,JOINT_ID,pinocchio::LOCAL,J);
-      const Eigen::VectorXd v     = -J.topRows<3>().bdcSvd(svdOptions).solve(err);
-      q = pinocchio::integrate(model,q,v*DT);
-      if(!(i%10)) std::cout << "error = " << err.transpose() << std::endl;
+      std::cout << "Convergence achieved!" << std::endl;
+      std::cout << "\nresult: " << q.transpose() << std::endl;
+      std::cout << "\nfinal error: " << err.transpose() << std::endl;
+      break;
     }
+    pinocchio::jointJacobian(model,data,q,JOINT_ID,J);
+    const Eigen::VectorXd v     = -J.topRows<3>().bdcSvd(svdOptions).solve(err);
+    q = pinocchio::integrate(model,q,v*DT);
+    if(!(i%10)) std::cout << "error = " << err.transpose() << std::endl;
 
-  if(i==IT_MAX)
-    std::cout << "\nWarning: the iterative algorithm has not reached convergence to the desired precision" << std::endl;
-
-  std::cout << "\nresult: " << q.transpose() << std::endl;
-  std::cout << "\nfinal error: " << err.transpose() << std::endl;
- 
+    if(i==IT_MAX)
```

Comment:
Thank you for fixing this example :)
This test will never work because it is now inside the for loop, therefore i will never be equal to IT_MAX inside the loop.
Aside from the obvious error, why did you change this example?
Also, any change to this file should be reflected in the corresponding Python example, and in the corresponding doc page [b-examples/i-inverse-kinematics.md](https://github.com/stack-of-tasks/pinocchio/blob/devel/doc/b-examples/i-inverse-kinematics.md), which reference the file line by line and is now (I think) broken

Reply:
`i==IT_MAX` can't be true inside the loop, that's right. But before, it was outside of the loop, and the test did not even compile, as `i` was defined inside the loop…

Reply:
The other parts I changed were changed for the very same reason: this file did not compile. A `;` was missing, and the call to `pinocchio::jointJacobian` was wrong, and the `err` which also defined in the loop was also accessed from outside of the loop.

Reply:
Damn, I don't know how it happened. Do as you please

Reply:
fixed in 53b7fdba6be2689a9c96f142c3da1b105b91a31d